apps/heft/src/configuration/HeftConfiguration.ts

```diff
@@ -24,6 +28,40 @@ export interface IHeftConfigurationInitializationOptions {
   terminalProvider: ITerminalProvider;
 }
 
+interface IProjectConfigurationFileEntry<TConfigFile> {
+  options: IProjectConfigurationFileSpecification<TConfigFile>;
+  loader: ProjectConfigurationFile<TConfigFile>;
+}
+
+const projectConfigurationFileRegistry: Map<string, IProjectConfigurationFileEntry<unknown>> = new Map();
+function getAndValidateProjectConfigurationFileEntry<TConfigFile>(
+  options: IProjectConfigurationFileSpecification<TConfigFile>
+): IProjectConfigurationFileEntry<TConfigFile> {
+  let entry: IProjectConfigurationFileEntry<TConfigFile> | undefined = projectConfigurationFileRegistry.get(
+    options.projectRelativeFilePath
+  ) as IProjectConfigurationFileEntry<TConfigFile> | undefined;
+
+  if (!entry) {
+    entry = {
+      options: Object.freeze(options),
+      loader: new ProjectConfigurationFile<TConfigFile>(options)
+    };
+  } else if (options !== entry.options) {
```

Comment:
Hmmmmm I'm not sure if this is a valid error. I think it's valid for multiple tools to try and load the same config file using different options. Yes, it would be better if projects would provide their config file via dependency hooks or something like that, but I'm not sure this check needs to exist. Ideally we should cache, keyed off of path + options, and get much of the same benefit here

Reply:
If we need to relax this later, we can.